nautobot/extras/api/serializers.py

```diff
@@ -766,6 +767,36 @@ class Meta:
         fields = "__all__"
 
 
+class ObjectMetadataSerializer(ValidatedModelSerializer):
+    assigned_object_type = ContentTypeField(
+        queryset=ContentType.objects.filter(FeatureQuery("metadata").get_query()),
+    )
+    assigned_object = serializers.SerializerMethodField()
+
+    class Meta:
+        model = ObjectMetadata
+        fields = "__all__"
+
+    @extend_schema_field(
+        PolymorphicProxySerializer(
+            component_name="ObjectMetadataAssignedObject",
+            resource_type_field_name="object_type",
+            serializers=lambda: nested_serializers_for_models(FeatureQuery("metadata").list_subclasses()),
```

Comment:
That's gotta be a pretty big set of serializers, isn't it? How bad is the impact on the schema?

Reply:
Implemented

Reply:
Unfortunately, it is a pretty big set of serializers. But it might be a necessary evil since this has always been our pattern for GFKs.

Reply:
Yeah, I think it's necessary - was just curious.